src/platform/Infineon/CYW30739/KeyValueStoreManagerImpl.cpp

```diff
@@ -149,8 +149,11 @@ CHIP_ERROR KeyValueStoreManagerImpl::_Delete(const char * key)
     err = CYW30739Config::ClearConfigValue(entry->GetKeyConfigKey());
     VerifyOrExit(ChipError::IsSuccess(err), err = CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
 
-    err = CYW30739Config::ClearConfigValue(entry->GetValueConfigKey());
-    VerifyOrExit(ChipError::IsSuccess(err), err = CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
+    if (entry->GetValueSize() != 0)
+    {
+        err = CYW30739Config::ClearConfigValue(entry->GetValueConfigKey());
+        VerifyOrExit(ChipError::IsSuccess(err), err = CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
+    }
```

Comment:
This will leave the key as present if the size is zero. How does this fix "network information does not restore back"? I think the fix likely needs to be in the NetworkCommissioning::WifiDriver instance for your platforn

Reply:
The subject of this PR contains misleading descriptions. It's a general KVS implementation fix.
When writing a zero-size value, CYW30739 only writes the key of the value to the flash because the lower layer APIs of CYW30739 doesn't support writing a zero-size value.
``` c
CHIP_ERROR KeyValueStoreManagerImpl::_Put(const char * key, const void * value, size_t value_size)
{
    ...
    if (value_size != 0)
    {
        SuccessOrExit(err = CYW30739Config::WriteConfigValueBin(entry->GetValueConfigKey(), value, value_size));
    }
    ...
}
```
CYW30739 only has to erase the key of a zero-size value from the flash rather than erase both the key and the value from the flash.
When the original codes tried to delete a non-existed value from the flash, the lower layer API would return an error and bypass the following codes. This is the bug we are trying to fix.

Reply:
This description should be in the issue linked to this PR.